python/lsst/pipe/analysis/colorAnalysis.py

```diff
@@ -363,281 +560,1040 @@ def plotGalaxyColors(self, catalogs, filenamer, dataId):
+
+                # Label total number of objects of each data type
+                lenNumObj = max(len(str(len(xColor[qaGood]))), len(str(len(xColor[inFitGood]))),
+                                len(str(len(xColor[inPerpGood]))))
+                fdx = max((min(0.09*lenNumObj, 0.9), 0.42))
+                xLoc, yLoc = xRange[0] + 0.03*deltaX, yRange[1] - 0.038*deltaY
+                kwargs = dict(va="center", fontsize=8)
+                axes.text(xLoc, yLoc, "NqaGood  =", ha="left", color="black", **kwargs)
+                axes.text(xLoc + fdx*deltaX, yLoc, str(len(xColor[qaGood])) + " [" + self.fluxFilter +
+                          " < " + str(self.config.analysis.magThreshold) + "]", ha="right", color="black",
+                          **kwargs)
+                yLoc -= 0.05*deltaY
+                axes.text(xLoc, yLoc, "NinFitGood =", ha="left", color="blue", **kwargs)
+                axes.text(xLoc + fdx*deltaX, yLoc, str(len(xColor[inFitGood])) + " [" + self.fluxFilter +
+                          " < " + str(self.config.analysis.magThreshold) + "]", ha="right", color="blue",
+                          **kwargs)
+                yLoc -= 0.05*deltaY
+                axes.text(xLoc, yLoc, "NinPerpGood =", ha="left", color="red", **kwargs)
+                axes.text(xLoc + fdx*deltaX, yLoc, str(len(xColor[inPerpGood])) + " [" + self.fluxFilter +
+                          " < " + str(self.config.analysis.magThreshold) + "]", ha="right", color="red",
+                          **kwargs)
+                if camera is not None:
+                    labelCamera(camera, plt, axes, 0.5, 1.09)
+                if hscRun is not None:
+                    axes.set_title("HSC stack run: " + hscRun, color="#800080")
+
+                tractStr = "tract: {:d}".format(dataId["tract"])
+                axes.annotate(tractStr, xy=(0.5, 1.04), xycoords="axes fraction", ha="center", va="center",
+                              fontsize=10, color="green")
+
+                fig.savefig(filename, dpi=120)
+                plt.close(fig)
+
+    def plotStarColorColor(self, principleColCats, byFilterCats, filenamer, dataId, fluxColumn, butler=None,
+                           camera=None, tractInfo=None, patchList=None, hscRun=None):
+        num = len(list(byFilterCats.values())[0])
         zp = 0.0
-        mags = {ff: zp - 2.5*np.log10(catalogs[ff][self.fluxColumn]) for ff in catalogs}
+        mags = {ff: zp - 2.5*np.log10(byFilterCats[ff][fluxColumn]) for ff in byFilterCats}
 
         bad = np.zeros(num, dtype=bool)
-        for cat in catalogs.values():
+        for cat in byFilterCats.values():
             for flag in self.flags:
                 if flag in cat.schema:
                     bad |= cat[flag]
 
         bright = np.ones(num, dtype=bool)
-        for mm in mags.values():
-            bright &= mm < self.config.analysis.magThreshold
+        bright &= mags[self.fluxFilter] < self.config.analysis.magThreshold
```

Comment:
`bright = mags[self.fluxFilter] < self.config.analysis.magThreshold` should do the trick?